### DISCUSSION TITLE
Error collecting test with --doctest-module

### DISCUSSION THREAD

**Discussion by ernestoarbitrio** [pytest-dev/pytest#8363] [Q&A]
Given that `pytest` works perfectly with all my unit and integration tests, i'm facing a problem dough when trying to run doctests with this command:
`pytest -svv tests/unit --doctest-modules`
The error seems related to a fixture (custom) i normally use in my tests, so I got this error:

```
PytestWarning: Got ValueError('no NPY fixture found at /mypath/tests/fixtures/mask-arrays/--wrapped--.npy') when unwrapping <tests.fixtures.LazyNPYResponseLoader object at 0x12f144760>.  This is usually caused by a violation of Python's object protocol; see e.g. https://github.com/pytest-dev/pytest/issues/5080
```
and then
```
FAILED tests/unit/test_util.py - ValueError: no NPY fixture found at /mypath/tests/fixtures/mask-arrays/--wrapped--.npy
```
Obviously that fixture doesn't exists cause the `--wrapped` file is the one generated from the doctest.py module.

I wonder if someone can give me a hint or tell me if I'm doing something wrong 🤷🏽‍♂️ 
Thx

**Comment by portyu9** [comment]
Hi @ernestoarbitrio 

The `--wrapped--.npy` lookup is a strong sign that your lazy object's `__getattr__` is treating Python's special `__wrapped__` attribute probe as one of your data keys.

Introspection code commonly checks `__wrapped__` when unwrapping decorated/callable objects. If your object does not provide that attribute, normal Python attribute semantics require `__getattr__` to raise `AttributeError`.

For example:

```python
def __getattr__(self, name):
    if name.startswith("__"):
        raise AttributeError(name)

    path = make_path_for(name)

    if not path.exists():
        raise AttributeError(name)

    return load(path)
```

In other words, arbitrary missing attributes should not turn into attempts to open files.

I would **not** create a `--wrapped--.npy` file or special-case doctest itself. The introspection is exposing an attribute-protocol issue in the custom lazy object.

Please mark this answer as accepted if it helped, thank you.
